File: page-objects/java-fluentlenium/complete/src/main/java/pages/ProductDetailPage.java
package pages;

import org.fluentlenium.core.FluentPage;
import org.fluentlenium.core.annotation.Page;
import org.openqa.selenium.By;

import java.util.concurrent.TimeUnit;

@SuppressWarnings("FieldCanBeLocal")
public class ProductDetailPage extends FluentPage {

    @SuppressWarnings("unused")
    @Page
    private ShoppingCartPage shoppingCartPage;

    public String getProductBrand() {
        return find(By.cssSelector("h2[class*='h-color-black'][class*='detail']")).first().text();
    }

    public String getProductName() {
        return find(By.cssSelector("h1[class*='h-text']")).first().text();
    }

    public void selectFirstAvailableSize() {
        find(By.cssSelector(".h-container.h-dropdown-placeholder")).click();
        String sizeSelector = "h5[class*='h-color-black'][class*='title-4'][class*='h-all-caps']";
        await().atMost(5, TimeUnit.SECONDS).until(el(sizeSelector)).present();
        find(By.cssSelector("h5[class*='h-color-black'][class*='title-4'][class*='h-all-caps']")).click();
    }

    public void addToShoppingCart() {
        find(By.cssSelector("#z-pdp-topSection-addToCartButton")).click();
    }

    public ShoppingCartPage goToShoppingCart() {
        String goToShoppingCartSelector = "a[class='z-navicat-header_navToolItemLink']";
        await().atMost(5, TimeUnit.SECONDS).until(el(goToShoppingCartSelector)).present();
        find(By.cssSelector("a[class='z-navicat-header_navToolItemLink']")).click();
        return shoppingCartPage;
    }
}

File: page-objects/js-webdriverio/complete/pageobjects/productDetail.page.js
// productDetail.page.js
var Page = require('./page');
var ShoppingCartPage = require('./shoppingCart.page');

var ProductDetailPage = Object.create(Page, {
    /**
     * define elements
     */
    sizeDropDown: { get: function () { return browser.element(".h-container.h-dropdown-placeholder"); } },
    firstAvailableSize: { get: function () { return browser.element("h5[class*='h-color-black'][class*='title-4'][class*='h-all-caps']"); } },
    addToShoppingCartButton: { get: function () { return browser.element("#z-pdp-topSection-addToCartButton"); } },
    goToShoppingCartButton: { get: function () {
        var shoppingCartButtonSelector = "a[class='z-navicat-header_navToolItemLink']";
        browser.waitForVisible(shoppingCartButtonSelector, 10000);
        return browser.element(shoppingCartButtonSelector);
    } },
    productBrand: { get: function () { return browser.element("h2[class*='h-color-black'][class*='detail']"); } },
    productName: { get: function () { return browser.element("h1[class*='h-text']"); } },

    getProductBrand: { value: function() {
            return this.productBrand.getText();
        } },

    getProductName: { value: function() {
            return this.productName.getText();
        } },

    clickOnFirstAvailableSize: { value: function() {
            this.sizeDropDown.click();
            this.firstAvailableSize.click();
        } },

    addToShoppingCart: { value: function() {
            this.addToShoppingCartButton.click();
        } },

    goToShoppingCart: { value: function() {
            this.goToShoppingCartButton.click();
            return ShoppingCartPage;
        } }
});

module.exports = ProductDetailPage;